src/tests/audio/tasks/forced_alignment_test.py

```diff
@@ -0,0 +1,215 @@
+        sentence_spans=None,
+    )
+
+
+@pytest.fixture
+def dummy_model() -> tuple:
+    """Fixture for a dummy model and processor."""
+    model_name = "facebook/wav2vec2-base-960h"
+    processor = Wav2Vec2Processor.from_pretrained(model_name)
+    model = Wav2Vec2ForCTC.from_pretrained(model_name)
+    return model, processor
+
+
+@pytest.fixture
+def script_line_fixture() -> ScriptLine:
+    """Pytest fixture to create a ScriptLine object.
+
+    Returns:
+        ScriptLine: An instance of ScriptLine.
+    """
+    data = {
+        "text": "test",
+        "speaker": "Speaker Name",
+        "start": 0.0,
+        "end": 10.0,
+        "chunks": [
+            {"text": "Chunk 1 text", "speaker": "Chunk 1 speaker", "start": 0.0, "end": 5.0},
+            {"text": "Chunk 2 text", "speaker": "Chunk 2 speaker", "start": 5.0, "end": 10.0},
+        ],
+    }
+    return ScriptLine.from_dict(data)
+
+
+def test_preprocess_segments() -> None:
+    """Test preprocessing of segments."""
+    transcript = [SingleSegment(start=0.0, end=1.0, text="test")]
+    model_dictionary = {"t": 0, "e": 1, "s": 2}
+    preprocessed_segments = _preprocess_segments(transcript, model_dictionary, "en", False, False)
+    assert preprocessed_segments[0]["clean_char"] == ["t", "e", "s", "t"]
+
+
+def test_can_align_segment(dummy_segment: SingleSegment) -> None:
+    """Test if a segment can be aligned."""
+    model_dictionary = {"t": 0, "e": 1, "s": 2}
+    assert _can_align_segment(dummy_segment, model_dictionary, 0.0, 10.0)
+
+
+def test_merge_repeats() -> None:
+    """Test merging of repeated tokens."""
+    path = [Point(0, 0, 1.0), Point(0, 1, 1.0), Point(1, 2, 1.0)]
+    transcript = "test"
+    segments = _merge_repeats(path, transcript)
+    assert len(segments) == 2
+
+
+def test_interpolate_nans() -> None:
+    """Test interpolation of NaN values."""
+    series = pd.Series([0.0, np.nan, 2.0])
+    interpolated_series = _interpolate_nans(series)
+    assert interpolated_series.isnull().sum() == 0
+
+
+if os.getenv("GITHUB_ACTIONS") != "true":
+
+    def test_get_prediction_matrix(dummy_model: tuple) -> None:
+        """Test generation of prediction matrix."""
+        model, _ = dummy_model
+        waveform_segment = torch.randn(1, 16000)
+        prediction_matrix = _get_prediction_matrix(model, waveform_segment, None, "huggingface", DeviceType.CPU)
+        assert prediction_matrix.shape[0] > 0
+
+    def test_align_segments(mono_audio_sample: Audio, dummy_model: tuple) -> None:
+        """Test alignment of segments."""
+        model, processor = dummy_model
+        model_dictionary = processor.tokenizer.get_vocab()
+
+        # Create a sample transcript
+        transcript = [SingleSegment(start=0.0, end=1.0, text="test")]
+
+        # Preprocess the transcript segments
+        preprocessed_transcript = _preprocess_segments(
+            transcript, model_dictionary, model_lang="en", print_progress=False, combined_progress=False
```

Comment:
why lang is a string?

Reply:
Because the main point of using the Language object was to validate the language code, which is done in the in interface function `align_transcriptions`, and I didn't have time to update it everywhere else. I just changed it though.